src/Nethermind/Nethermind.Network/SnapCapabilitySwitcher.cs

```diff
@@ -27,29 +28,30 @@ namespace Nethermind.Network;
 public class SnapCapabilitySwitcher
 {
     private readonly IProtocolsManager _protocolsManager;
-    private readonly ProgressTracker _progressTracker;
+    private readonly MultiSyncModeSelector _multiSyncModeSelector;
+    private readonly ILogger _logger;
 
-    public SnapCapabilitySwitcher(IProtocolsManager? protocolsManager, ProgressTracker? progressTracker)
+    public SnapCapabilitySwitcher(IProtocolsManager? protocolsManager, MultiSyncModeSelector? multiSyncModeSelector, ILogManager? logManager)
     {
         _protocolsManager = protocolsManager ?? throw new ArgumentNullException(nameof(protocolsManager));
-        _progressTracker = progressTracker ?? throw new ArgumentNullException(nameof(progressTracker));
+        _multiSyncModeSelector = multiSyncModeSelector ?? throw new ArgumentNullException(nameof(multiSyncModeSelector));
+        _logger = logManager.GetClassLogger() ?? throw new ArgumentNullException(nameof(logManager));
     }
 
     /// <summary>
     /// Add Snap capability if SnapSync is not finished and remove after finished.
     /// </summary>
     public void EnableSnapCapabilityUntilSynced()
     {
-        if (!_progressTracker.IsSnapGetRangesFinished())
-        {
-            _protocolsManager.AddSupportedCapability(new Capability(Protocol.Snap, 1));
-            _progressTracker.SnapSyncFinished += OnSnapSyncFinished;
-        }
+        _protocolsManager.AddSupportedCapability(new Capability(Protocol.Snap, 1));
+        _multiSyncModeSelector.SnapSyncFinished += OnSnapSyncFinished;
+        if (_logger.IsDebug) _logger.Debug("Enabled snap capability");
     }
 
     private void OnSnapSyncFinished(object? sender, EventArgs e)
     {
-        _progressTracker.SnapSyncFinished -= OnSnapSyncFinished;
+        _multiSyncModeSelector.SnapSyncFinished -= OnSnapSyncFinished;
         _protocolsManager.RemoveSupportedCapability(new Capability(Protocol.Snap, 1));
+        if (_logger.IsWarn) _logger.Warn("State sync finished. Disabled snap capability.");
```

Comment:
Do we want to have this as a warning?

Reply:
I wanted to have it visible - it's very important log. I can change to `Info` or even `Debug` (but would prefer to have it at least at `Info` level to be included in default logs)

Reply:
I don't think we should use Warnings in prod code for notifying about expected events even if they're important.
It would be good to add smth like **Important** logging level :)

Reply:
https://github.com/NethermindEth/nethermind/issues/4675

Reply:
We are already using `Warn` for notifying about expected events, like:
`EthStatsPlugin plugin disabled due to EthStatsConfig settings set to false`
`Starting Clique block producer & sealer`
But I agree it is not a good way and creating `Important` level (between `Info` and `Warn`) is a great idea!
For now I will change this log to just `Info`